### ISSUE TITLE
Klarheit bzgl. "Übersetzung von Enden" bei `FindSupportST`

### ISSUE THREAD

**Issue by jvoigtlaender** [fmidue/modelling-tasks#232]
Angesichts von 
![image](https://github.com/jvoigtlaender/modelling-tasks-internal/assets/5853832/af1138f8-4430-40c8-8dc4-4a358a6201d8)
sollte eventuell bei `FindSupportST` immer `activityFinalsExist = Just False` gesetzt sein.

Aber auch schon nur bezüglich 
![image](https://github.com/jvoigtlaender/modelling-tasks-internal/assets/5853832/d461fb57-2410-486c-bf83-246c73d1764f)
ist gewissermaßen problematisch, dass die Folien keine genaue Übersetzungsregel angeben (anders als für alle anderen AD-Elemente).

In den Video-Untertiteln steht dazu: https://github.com/fmidue/Modellierung-Untertitel/blob/3f4208addd2cffac7dafe60d80ec69ee60c6d730/Vorlesungen/Woche11/Aktivitaetsdiagramme_-_Elemente_und_Uebersetzung_zu_PNs_korrigiert.srt#L1616-L1634

In einem konkreten Beispiel habe ich es dann so gemacht: https://github.com/fmidue/Modellierung-Untertitel/blob/3f4208addd2cffac7dafe60d80ec69ee60c6d730/Vorlesungen/Woche11/Aktivitaetsdiagramme_-_Elemente_und_Uebersetzung_zu_PNs_korrigiert.srt#L1735-L1753

Aber es klingt dennoch durch, dass es prinzipiell verschiedene Möglichkeiten gibt, die Enden zu übersetzen.

Bei den anderen Aufgaben ist das nicht so problematisch, weil es sich nicht auswirkt oder man etwa bei `PetriMatch` schon eine konkrete Übersetzung als Petrinetz vorgegeben kriegt. Das sollte wohl (fast) immer eventuelle Mehrdeutigkeiten verhindern. Aber bei `FindSupportST` muss man ja selbst die Übersetzung machen, und je nachdem, wie man es genau macht, kommen natürlich andere Anzahlen von Elementen im Petrinetz heraus.

Wahrscheinlich könnte und müsste man dem durch Erklärtext begegnen. Also etwas in die Richtung sagen, dass Flussenden (mal angenommen, Aktivitätsenden schließen wir sowieso aus) nicht durch zusätzliche Stellen realisiert werden sollen und dass nach Ablauf des (übersetzten) Diagramms keine Marken mehr im Petrinetz sein sollen. Zumindest scheint mir im Moment, dass damit die Übersetzung und folglich auch die Zählung für die Beantwortung der Aufgabe eindeutig würde.

**Comment by jvoigtlaender** [comment]
Ein klein bisschen Auswirkung könnte es auch noch auf `SelectPetri`-Aufgaben geben, in dem vielleicht sehr seltenen Fall, dass neben dem "korrekten Petrinetz" zufällig auch noch ein "inkorrekt gemeintes Petrinetz" vorkommt, welches aber unter einer abweichenden Interpretation des Übersetzens von Aktivitäts- oder Flussenden doch auch korrekt sein könnte.

**Comment by marcellussiegburg** [comment]
Bei `MatchPetri` gibt es die Möglichkeit, die Option `noActivityFinalInForkBlocks = Just True` zu setzen (diese sollte dort vielleicht auch forciert werden, wenn Aktivitätsenden und auch Flussenden erlaubt sind?).

Vielleicht würde das Einführen und überlegte forcieren dieser Option (anstatt eines kompletten Verbots) auch hier helfen (wahrscheinlich mit entsprechendem Erklärtext)?

**Comment by jvoigtlaender** [comment]
Ja, ergibt Sinn. Über die genaue Ausgestaltung, wann was verboten ist, kann noch abgewogen werden. Zum Beispiel, ob es auch bei `SelectPetri` passieren sollte.

**Comment by marcellussiegburg** [comment]
`noActivityFinalInForkBlocks = Just True` reicht vielleicht auch noch nicht aus? Weil damit trotzdem das Aktivitätsende nicht alle Flüsse beendet.

**Comment by jvoigtlaender** [comment]
> `noActivityFinalInForkBlocks = Just True` reicht vielleicht auch noch nicht aus? Weil damit trotzdem das Aktivitätsende nicht alle Flüsse beendet.

Was wäre eine entsprechende problematische Situation? Wenn innerhalb des Fork-Blocks irgendwie noch ein "loser Fluss" existiert, der nicht zum Join-Knoten gezwungen wurde?

**Comment by marcellussiegburg** [comment]
> Wenn innerhalb des Fork-Blocks irgendwie noch ein "loser Fluss" existiert, der nicht zum Join-Knoten gezwungen wurde?

Ja, genau. Es könnte sein, dass dort noch eine Aktivität vor einem Flussende steht, aber ganz am Ende (irgendwo nach dem Join) ein Aktivitätsende kommt.

**Comment by jvoigtlaender** [comment]
Dann sollte bei `FindSupportST` tatsächlich immer `activityFinalsExist = Just False` gesetzt werden, um diesen Teil des Problems zu lösen.

**Comment by jvoigtlaender** [comment]
Ist das nicht alles schon dadurch abgedeckt, dass bereits generell verboten wird (bei Aufgaben mit AD-PN-Übersetzung), dass es sowohl ein Aktivitätsende als auch noch mindestens ein Flussende gibt?

**Comment by marcellussiegburg** [comment]
> Ist das nicht alles schon dadurch abgedeckt, dass bereits generell verboten wird (bei Aufgaben mit AD-PN-Übersetzung), dass es sowohl ein Aktivitätsende als auch noch mindestens ein Flussende gibt?

Ja, das deckt das nun ab.
